src/sympc/protocol/falcon/falcon.py

```diff
@@ -63,3 +65,87 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return True
+
+    @staticmethod
+    def mul_master(x: MPCTensor, y: MPCTensor, session: Session) -> MPCTensor:
+        """Master method for multiplication.
+
+        Performs Falcon's mul implementation, gets and reshares mul results and distributes shares.
+        This needs to be improved in future, it relies on orchestrator being a trusted third party.
+        Falcon, requires parties to be able to communication between each other.
+
+        Args:
+            x (MPCTensor): Secret
+            y (MPCTensor): Another secret
+            session (Session): Session the tensors belong to
+
+        Returns:
+            shares (ReplicatedSharedTensor): Shares in terms of ReplicatedSharedTensor.
+
+        Raises:
+            ValueError: Raised when number of parties are not three.
+            NotImplementedError: Raised when implementation not present
+
+        """
+        if len(session.parties) != 3:
+            raise ValueError("Falcon requires 3 parties")
+
+        result = None
+
+        if session.protocol.security_type == "semi-honest":
+            result = Falcon.multiplication_protocol(x, y)
+
+        elif session.protocol.security_type == "malicious":
+            raise NotImplementedError(
+                "Multiplication has not been implemented for malicious setting"
+            )
+        else:
+            raise NotImplementedError(
+                f"mult operation not implemented for {session.protocol.security_type} setting"
+            )
+
+        przs_masks = [
+            session.session_ptrs[index]
+            .przs_generate_random_share(shape=x.shape)
+            .get_shares()
```

Comment:
MAYBE: Can't we do first `[0]` and then `get`? (it might be faster ?)

Reply:
Yeah, I had tried this first. 
przs_generate_random_share returns Anypointer which isn't subscriptable throwing an error:

`FAILED falcon_test.py::test_mul_private_integer - TypeError: 'AnyPointer' object is not subscriptable`